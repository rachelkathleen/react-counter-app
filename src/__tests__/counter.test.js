import Counter from '../counter';
import React from 'react';
import { shallow } from 'enzyme';

describe('counter', () => {
  let beginningCount;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Counter />);
    beginningCount = wrapper.instance().state.count;
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

<<<<<<< HEAD
  it('adds to the count when the + button is clicked', () => {
=======
  it('adds to the count when the + button is pressed', () => {
>>>>>>> add jest and enzyme
    wrapper.instance().incrementCount();

    expect(wrapper.instance().state.count).toBe(beginningCount + 1);
  });

<<<<<<< HEAD
  it('subtracts to the count when the - button is clicked', () => {
=======
  it('subtracts from the count when the - button is pressed', () => {
>>>>>>> add jest and enzyme
    wrapper.instance().decrementCount();

    expect(wrapper.instance().state.count).toBe(beginningCount - 1);
  });
});
