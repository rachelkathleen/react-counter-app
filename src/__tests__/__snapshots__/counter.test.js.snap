// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter renders correctly 1`] = `
<div>
  <div
    className="count"
  >
    <h3>
      Count:
    </h3>
    <h1>
      0
    </h1>
  </div>
  <div
    className="buttons"
  >
    <Button
      action={[Function]}
      id="plus"
      title="-"
    />
    <Button
      action={[Function]}
      id="minus"
      title="+"
    />
  </div>
</div>
`;
